torch/_inductor/scheduler.py

```diff
@@ -2105,6 +2169,45 @@ def add_user(
             for buf in node.get_outputs():
                 buf.set_users(name_to_users[buf.get_name()].items)
 
+        if config.reorder_for_peak_memory:
+            # copy user information onto freeable input buffers
+            for input_buf_name, input_buf in self.name_to_input_buf.items():
+                input_buf.set_users(name_to_users[input_buf_name].items)
+
+    def compute_size_for_scheduler_buffer(self) -> None:
```

Comment:
I think it could be worth moving all the new functions to a dedicated file (e.g. `memory.py` in the same folder), to make the `scheduler.py` file shorter and also provide an easy place to find all the memory passes.